x-pack/plugins/cross_cluster_replication/public/app/store/actions/follower_index.js

```diff
@@ -39,25 +45,33 @@ export const loadFollowerIndices = (isUpdating = false) =>
 export const getFollowerIndex = (id) =>
   sendApiRequest({
     label: t.FOLLOWER_INDEX_GET,
-    scope,
+    scope: `${scope}-get`,
     handler: async () => (
       await getFollowerIndexRequest(id)
     )
   });
 
-export const saveFollowerIndex = (name, followerIndex) => (
+export const saveFollowerIndex = (name, followerIndex, isUpdating) => (
```

Comment:
I like how we're unifying the behavior of the create and update actions -- that's really nice for maintainability! I think there's a way we can simplify both the interface and implementation.

In the past I've found that conditional flags like `isUpdating` tend to bloat the function signature, make the call sites more difficult to follow, and also add complexity to the implementation in the form of with parallel branches split across multiple conditions.

I think we can simplify the code and make it clearer by exporting separate functions for create and update, and extracting the common logic into an internal function. Does this seem like an improvement to you?

```js
const saveFollowerIndex = (name, followerIndex, handler, successMessage) => (
  sendApiRequest({
    label: t.FOLLOWER_INDEX_CREATE,
    status: API_STATUS.SAVING,
    scope: `${scope}-save`,
    handler,
    onSuccess() {
      toastNotifications.addSuccess(successMessage);
      routing.navigate(`/follower_indices`, undefined, {
        pattern: encodeURIComponent(name),
      });
    },
  })
);

export const createFollowerIndex = (name, followerIndex) => (
  const handler = async () => {
    return await createFollowerIndexRequest({ name, ...followerIndex });
  };

  const successMessage = i18n.translate('xpack.crossClusterReplication.followerIndex.addAction.successNotificationTitle', {
    defaultMessage: `Added follower index '{name}'`,
    values: { name },
  });

  saveFollowerIndex(name, followerIndex, handler, successMessage);
);

export const updateFollowerIndex = (name, followerIndex) => (
  const handler = async () => {
    return await updateFollowerIndexRequest(name, followerIndex);
  };

  const successMessage = i18n.translate('xpack.crossClusterReplication.followerIndex.updateAction.successNotificationTitle', {
    defaultMessage: `Follower index '{name}' updated successfully`,
    values: { name },
  });

  saveFollowerIndex(name, followerIndex, handler, successMessage);
);
```

Reply:
I am deferring this for now since both follower index and auto follow pattern actions perform create and update with one save method, and I don't want to introduce any regressions this late 😄